src/acvm_interop/smart_contract.rs

```diff
@@ -75,47 +75,50 @@ impl SmartContract for Barretenberg {
     }
 }
 
-#[test]
-fn test_smart_contract() -> Result<(), BackendError> {
-    use crate::barretenberg_structures::{Constraint, ConstraintSystem};
-    use crate::composer::Composer;
-    use crate::Barretenberg;
-    use acvm::FieldElement;
-    use tokio::runtime::Builder;
-
-    let constraint = Constraint {
-        a: 1,
-        b: 2,
-        c: 3,
-        qm: FieldElement::zero(),
-        ql: FieldElement::one(),
-        qr: FieldElement::one(),
-        qo: -FieldElement::one(),
-        qc: FieldElement::zero(),
-    };
-
-    let constraint_system = ConstraintSystem::new()
-        .var_num(4)
-        .public_inputs(vec![1, 2])
-        .constraints(vec![constraint]);
-
-    let bb = Barretenberg::new();
-    let crs = Builder::new_current_thread()
-        .enable_all()
-        .build()
-        .unwrap()
-        .block_on(bb.get_crs(&constraint_system))?;
-
-    let proving_key = bb.compute_proving_key(&constraint_system)?;
-    let verification_key = bb.compute_verification_key(&crs, &proving_key)?;
-
-    let common_reference_string: Vec<u8> = crs.try_into()?;
-
-    let contract = bb.eth_contract_from_vk(&common_reference_string, &verification_key)?;
-
-    assert!(contract.contains("contract BaseUltraVerifier"));
-    assert!(contract.contains("contract UltraVerifier"));
-    assert!(contract.contains("library UltraVerificationKey"));
-
-    Ok(())
+#[cfg(test)]
+mod tests {
+    use acvm::SmartContract;
+    use tokio::test;
+
+    use crate::BackendError;
+
+    #[test]
```

Comment:
```suggestion
    use acvm::SmartContract;

    use crate::BackendError;

    #[tokio::test]
```

Not going to block on this but I prefer the qualified usage of this to make is clearer that we're running this test inside tokio. I can follow up with this change in a separate PR however if you don't have strong feelings on why you did it this way.

Reply:
I purposefully imported it that way, because I will use conditional compilation to `use wasm_bindgen_test::wasm_bindgen_test as test` and then the macros on each don't need to be duplicated nor do we need to create separate tests.

Reply:
Gotcha, yeah I noticed this would make it easier for conditional compilation which was the main reason I didn't go ahead with the change.